test/test-command.el

```diff
@@ -156,4 +156,49 @@ resource \"elasticstack_elasticsearch_security_user\" \"filebeat_writer\" {
     (cl-letf (((symbol-function 'terraform--get-resource-provider-namespace) (lambda (prov) "elastic")))
       (should (equal (terraform--resource-url-at-point) "https://registry.terraform.io/providers/elastic/elasticstack/latest/docs/resources/elasticsearch_security_user")))))
 
+(ert-deftest command--terraform--get-resource-provider-source-in-buffer ()
+  (with-terraform-temp-buffer
+   "
+# blah blah
+terraform {
+  required_providers {
+    aws = {
+      source  = \"hashicorp/aws\"
+      version = \"~> 5\"
+    }
+    azurerm = {
+      source  = \"hashicorp/azurerm\"
+      version = \"~> 3.47\"
+    }
+  }
+}
+"
+   (should (equal (terraform--get-resource-provider-source-in-buffer "azurerm") "hashicorp/azurerm"))
+   ;;(should (equal (terraform--get-resource-provider-source-in-buffer "plop") nil))
+   (should (equal (terraform--get-resource-provider-source-in-buffer "aws") "hashicorp/aws"))))
+
+;; required_providers is defined in current buffer
+(ert-deftest command--terraform--get-resource-provider-source ()
+  (with-terraform-temp-buffer
+   "
+# blah blah
+terraform {
+  required_providers {
+    aws = {
+      source  = \"hashicorp/aws\"
+      version = \"~> 5\"
+    }
+    azurerm = {
+      source  = \"hashicorp/azurerm\"
+      version = \"~> 3.47\"
+    }
+  }
+}
```

Comment:
Here we should probably put some different content so it really loads the data from the file and not from the buffer (as a fallback).

Reply:
Loading provider source from this file is done in test `command--terraform--get-resource-provider-source-provider-in-file` line 201.